cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -126,38 +126,41 @@ func NewFromCluster(client clientset.Interface, certificatesDir string) (*Client
 	// The first case should be dropped in v1.14 when support for v1.12 clusters can be removed from the codebase.
 
 	// Detect which type of etcd we are dealing with
+	// Please note that this test can be executed only on master nodes during upgrades;
+	// For nodes where we are joining a new control plane node instead we should tolerate that the etcd manifest does not
+	// exists and try to connect to etcd using API server advertise address; as described above this will lead to a know isse
+	// for cluster created with v1.12, but a documented workaround will be provided
 	oldManifest := false
 	klog.V(1).Infoln("checking etcd manifest")
 
 	etcdManifestFile := constants.GetStaticPodFilepath(constants.Etcd, constants.GetStaticPodDirectory())
 	etcdPod, err := staticpod.ReadStaticPodFromDisk(etcdManifestFile)
-	if err != nil {
-		return nil, errors.Wrap(err, "error reading etcd manifest file")
```

Comment:
i think we still need to have the `err != nil` case.

Reply:
The fact is that in case join --control-plane we have an error by design (because the etcd pod doesn't exist)

Ps. I consider acceptable this solution based on parsing manifests *only* because this is something that should go away the next release

Reply:
ok, understood. `etcdPod` is also not used for the `err != nil` case, so this should be fine.
/lgtm